### ISSUE TITLE
Field Type extension notice from configuration-notice

### ISSUE THREAD

**Issue by LordSimal** [bolt/core#2615]
Details
-------

| Question                      | Answer
|----------------- | ------
| Relevant Bolt Version | 4.1.20
| Install type                   |  Composer install
| BC Break                      | no
| PHP version                 | 7.4
| Web server                  | Nginx

Reproduction
------------

* Install current BoltCMS 4 via composer + demo data
* Install https://github.com/LordSimal/bolt-geolocation-field
* Add test content-type + geolocation field
* See dashboard notice on the right

<img width="1792" alt="2021-06-06_18-05-19" src="https://user-images.githubusercontent.com/9105243/120931433-c9cceb00-c6f1-11eb-90e7-3c18b4355a74.png">

Basically I already did some debugging in https://github.com/LordSimal/bolt-geolocation-field/issues/7 but couldn't get to the core because it has something to do with doctrine magic.

Bob said it probably has something to do with the core, so here is that issue 😄

**Comment by I-Valchev** [comment]
Hey @LordSimal ,

You are right, the doctrine magic happens here:
https://github.com/bolt/core/blob/4.1/src/Repository/FieldRepository.php#L138-L159

It's looking for `GeolocationField` that implements the `Bolt\Entity\FieldInterface`, which you already do here:
https://github.com/LordSimal/bolt-geolocation-field/blob/master/src/Entity/GeolocationField.php

Are you able to make some `dump` here in there in the `getFieldClassname` method, to investigate at which point it goes unexpectedly? Also, make sure that you're using a recent version of the `configuration-notices` extension, because some time ago it did not support identifying non-core fields.

Happy to help out on Slack, if you want to discuss :-)

**Comment by LordSimal** [comment]
Hi Ivo,

I nailed it down to the fact, that `self::$em->getMetadataFactory()->getAllMetadata()` doesn't contain my entity class `GeolocationField` which does implement `Bolt\Entity\FieldInterface`

Why it doesn't find it, I have no idea.

**Comment by LordSimal** [comment]
Following these methods I get to 

`/vendor/doctrine/persistence/lib/Doctrine/Persistence/Mapping/AbstractClassMetadataFactory.php` function `getAllMetadata()`

which calls `$driver->getAllClassNames()`
`/vendor/doctrine/persistence/lib/Doctrine/Persistence/Mapping/Driver/MappingDriverChain.php` function `getAllClassNames()`

which then foreaches through its drivers via `$this->drivers`

And as far as I have seen drivers are being added via the generated cache in 
`var/cache/dev/ContainerXYZ/App_KernelDevDebugContainer.php`
![image](https://user-images.githubusercontent.com/9105243/121042580-1e389f00-c7b4-11eb-8e81-28a74ad3cc1b.png)

But when I generate the cache my entity class doesn't get generated in there, therefore missing in the drivers etc. etc.

**Comment by LordSimal** [comment]
renaming the `config/services.yml` to `config/services.yaml` did indeed fix the notice.
I have no idea why but I guess its some crazy Symfony magic.

**Comment by I-Valchev** [comment]
You're welcome! Thanks for fixing it in the extension so speedily.

Here's the fix: https://github.com/bolt/core/pull/2618
Essentially, Bolt was only copying the config files ending up in `.yaml`. That PR allows both `.yaml` and `.yml` :-)
